server/http/api/api.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/configurator"
+	"github.com/mattermost/mattermost-plugin-apps/server/constants"
+
+	"github.com/gorilla/mux"
+
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
+	"github.com/mattermost/mattermost-plugin-apps/server/apps"
+)
+
+const (
+	SubscribePath = "/subscribe"
+)
+
+type SubscribeResponse struct {
+	Error  string            `json:"error,omitempty"`
+	Errors map[string]string `json:"errors,omitempty"`
+}
+
+type api struct {
+	mm   *pluginapi.Client
+	apps *apps.Service
+	// subs         *apps.Subscriptions
+	configurator configurator.Service
+}
+
+func Init(router *mux.Router, apps *apps.Service) {
+	a := api{
+		mm:           apps.Mattermost,
+		configurator: apps.Config,
+	}
+
+	subrouter := router.PathPrefix(constants.APIPath).Subrouter()
+	subrouter.HandleFunc(SubscribePath, a.handleSubscribe).Methods("POST", "DELETE")
```

Comment:
0/5 not sure what the Gorilla-idiomatic style is, but seems like having separate handlers defined for the methods would be cleaner. The bolierplate of handling requests can be worked into helper function(s).

Reply:
I picked this implementation because the endpoint is the same and the function is determined by a switch statement based on the method type.  I don't see many variations of `subscribe` path so chose this instead of two handlers (handleSubscribeStore, handleSubscribeDelete).  I'm also not positive if this is idiomatic Gorilla. I just saw the option to pass in a string slice